pwnlib/log.py

```diff
@@ -185,7 +185,7 @@ def __init__(self, logger, msg, status, level, args, kwargs):
 
     def _log(self, status, args, kwargs, msgtype):
         # Logs are strings, not bytes.  Handle Python3 bytes() objects.
-        status = six.ensure_text(status)
```

Comment:
Should we break out _need_text and friends into their own module?  `context` seems like the wrong place for the code, maybe for storing the encoding but not the rest of the tools.

Separately, why was `ensure_text` insufficient here?

Reply:
`ensure_text` does not raise any warnings nor exceptions

Reply:
Ah the new stuff raises warnings / exceptions? Neat!